pkg/queue/stats_reporter.go

```diff
@@ -0,0 +1,164 @@
+// TODO(@mrmcmuffinz): Need to get more appropriate names.
+const (
+	// ReportingPeriod interval of time for reporting.
+	ReportingPeriod = 10 * time.Second
+	// RequestCountM number of requests received since last Stat (approximately QPS).
+	RequestCountM Measurement = iota
+	// AverageConcurrentRequestsM average number of requests currently being handled by this pod.
+	AverageConcurrentRequestsM
+	// LameDuckM indicates this Pod has received a shutdown signal.
+	LameDuckM
+)
+
+// TODO(@mrmcmuffinz): Need to move this to a more appropriate place.
+var (
+	measurements = []*stats.Float64Measure{
+		RequestCountM: stats.Float64(
+			"request_count",
+			"Number of requests received since last Stat",
+			stats.UnitNone),
+		AverageConcurrentRequestsM: stats.Float64(
+			"average_concurrent_requests",
+			"Number of requests currently being handled by this pod",
+			stats.UnitNone),
+		LameDuckM: stats.Float64(
+			"lame_duck",
+			"Indicates this Pod has received a shutdown signal with 1 else 0",
+			stats.UnitNone),
+	}
+	namespaceTagKey tag.Key
+	configTagKey    tag.Key
+	revisionTagKey  tag.Key
+	serviceTagKey   tag.Key
+)
+
+func init() {
+	var err error
+	// Create the tag keys that will be used to add tags to our measurements.
+	// Tag keys must conform to the restrictions described in
+	// go.opencensus.io/tag/validate.go. Currently those restrictions are:
+	// - length between 1 and 255 inclusive
+	// - characters are printable US-ASCII
+	namespaceTagKey, err = tag.NewKey("configuration_namespace")
+	if err != nil {
+		panic(err)
+	}
+	serviceTagKey, err = tag.NewKey("service")
+	if err != nil {
+		panic(err)
+	}
+	configTagKey, err = tag.NewKey("configuration")
+	if err != nil {
+		panic(err)
+	}
+	revisionTagKey, err = tag.NewKey("revision")
+	if err != nil {
+		panic(err)
+	}
+
+	// Create views to see our measurements. This can return an error if
+	// a previously-registered view has the same name with a different value.
+	// View name defaults to the measure name if unspecified.
+	err = view.Register(
+		&view.View{
+			Description: "Number of requests received since last Stat",
+			Measure:     measurements[RequestCountM],
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{namespaceTagKey, serviceTagKey, configTagKey, revisionTagKey},
+		},
+		&view.View{
+			Description: "Number of requests currently being handled by this pod",
+			Measure:     measurements[AverageConcurrentRequestsM],
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{namespaceTagKey, serviceTagKey, configTagKey, revisionTagKey},
+		},
+		&view.View{
+			Description: "Indicates this Pod has received a shutdown signal with 1 else 0",
+			Measure:     measurements[LameDuckM],
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{namespaceTagKey, serviceTagKey, configTagKey, revisionTagKey},
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+}
+
+/*
+// TODO(@mrmcmuffinz): Need to figure out how to properly do the reporting.
```

Comment:
This all looks totally reasonable so far.

To do the actual reporting you might call `reporter.Report(AverageConcurrentRequestsM, avg)` about here: https://github.com/knative/serving/blob/917af9cf3306e9d15a0498af78ed0f7e3450ad04/pkg/queue/stats.go#L119 where we throw the stat into our old pipeline.

Reply:
Thanks, will take a look at.